src/components/top-bar/styles.ts

```diff
@@ -0,0 +1,87 @@
+import { css } from '@linaria/core'
+import { styled } from '@linaria/react'
+import { isDesktopOrBelow, isMobile, isTabletOrBelow } from '../../styles/media'
+import { ElButtonGroup } from '../button-group'
+import { ElNavSearchButton } from '../nav-search-button/styles'
+import { ElButtonNavIconItem } from '../nav-icon-item'
+
+export const elTopBarAvatar = css`
+  display: flex;
+  padding: var(--spacing-1, 4px) var(--spacing-none, 0px);
+  padding-left: var(--spacing-2, 8px);
+  align-items: center;
+  gap: 10px;
+`
+
+export const ElTopBarMainNav = styled(ElButtonGroup)`
+  display: flex;
+  padding: var(--spacing-1, 4px) var(--spacing-none, 0px) var(--spacing-1, 4px) var(--spacing-6, 24px);
+  align-items: flex-start;
+  flex-grow: 1;
+`
+
+export const ElTopBarSecondaryNav = styled(ElButtonGroup)`
+  flex-wrap: nowrap;
+`
+
+export const ElTopBar = styled.div`
+  display: flex;
+  height: var(--size-10, 56px);
+  padding: var(--spacing-2, 8px) var(--spacing-5, 20px);
+  justify-content: space-between;
+  align-items: center;
+  flex-shrink: 0;
+  border-bottom: var(--border-default, 1px) solid var(--outline-default, #e5e9ed);
+  background: var(--fill-white, #fff);
+
+  ${isDesktopOrBelow} {
+    padding: var(--spacing-2, 8px) var(--spacing-4, 16px);
+
+    // TODO: need to be adjusted once AppSwitcher and Brand/Logo included
+    justify-content: flex-end;
+
+    ${ElTopBarMainNav}, ${ElTopBarSecondaryNav} {
+      display: none;
+    }
+  }
+
+  ${isTabletOrBelow} {
+    ${elTopBarAvatar} {
+      display: none;
+    }
+  }
+`
+
+export const ElTopBarSearchContainer = styled.div`
+  padding: var(--spacing-1, 4px) var(--spacing-4, 16px) var(--spacing-1, 4px) var(--spacing-none, 0px);
+  ${ElNavSearchButton} {
+    width: 216px;
+  }
+
+  ${isMobile} {
+    padding-right: var(--spacing-spacing-2, 8px);
+
+    ${ElNavSearchButton} {
```

Comment:
**issue:** again, we don't need to make the `ElNavSearchButton` `display: none`, rather we can make `ElTopBarSearchContainer` `display: none` and ensure our behaviour (not displaying at a certain screen size) does not depend on the specific component a consumer places inside `ElTopBarSearchContainer`.

Reply:
actually `ElTopBarSearchContainer` have 2 child, which search button will display in big screen, and search icon for small screen,

```javascript
  <TopBar.Search>
     <NavIconItem icon={<Icon icon="search" />} aria-label="nav-icon-item-example" />
     <NavSearchButton />
  </TopBar.Search>
```

Reply:
Ah yes, my apologies. I think the evidence for making `NavSearchButton` take care of the button to nav icon item change at the mobile is mounting 🤔

This would simplify the above to:
```
<TopBar.Search>
  <NavSearchButton />
</TopBar.Search>
```

And `TopBar.Search` would only need to be adjusting it's right padding. What do you think?

Reply:
> And TopBar.Search would only need to be adjusting it's right padding. What do you think?

yes, I think thats simpler, and then `<NavIconItem icon={<Icon icon="search" />} />` will be part of `NavSearchButton`  right?

Reply:
Yes, which I think may have been something you had originally suggested 😅 If so, apologies for taking a while to see the value haha

Reply:
No worries, I'm glad we're on the same page now. 😄